Mathlib/Data/Finsupp/Weight.lean

```diff
@@ -197,7 +215,7 @@ theorem finite_of_nat_weight_le [Finite σ] (w : σ → ℕ) (hw : ∀ x, w x 
 end CanonicallyOrderedAddCommMonoid
 
 /-- The degree of a finsupp function. -/
-def degree (d : σ →₀ ℕ) := ∑ i ∈ d.support, d i
+def degree {N : Type*} [AddCommMonoid N] (d : σ →₀ N) := ∑ i ∈ d.support, d i
```

Comment:
It looks like a bunch more lemmas in the API generalise. Can you go through it all?

Reply:
I tried to. A few lemmas are still restricted to Nat.